tools/topology/sof-tgl-nocodec-8ch.m4

```diff
@@ -0,0 +1,186 @@
+#
+# Topology for generic Tigerlake board with no codec and digital mic array.
+#
+# TGL Host GW DMAC support max 6 playback and max 6 capture channels so some
+# pipelines/PCMs/DAIs are commented out to keep within HW bounds. If these
+# are needed then they can be used provided other PCMs/pipelines/SSPs are
+# commented out in their place.
```

Comment:
this is really awful to review. You should use one file and macros to select the configuration you want.

Reply:
@xiulipan also commented out code doesn't get compiled. So, if possible, better make this compile-time selectable without the need to edit. Then it can also be included in CI.